compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -96,6 +97,79 @@ static bool getTilingOptionsFromConfig(func::FuncOp funcOp, int64_t tilingLevel,
   return false;
 }
 
+/// Computes the canonical shape used to vectorize this dispatch. Retrieves
+/// the vectorization tile sizes (parallel and reduction levels) out of the
+/// lowering config and adjusts them to the format expected by the Linalg
+/// vectorizer.
+static SmallVector<int64_t> getCanonicalVectorShape(func::FuncOp funcOp) {
+  FailureOr<Operation *> rootOp = getRootOp(funcOp);
+  if (failed(rootOp)) {
+    return {};
+  }
+
+  unsigned numTileLevels =
+      mlir::iree_compiler::getNumTileLevels(rootOp.value());
+  if (numTileLevels < 3) {
+    return {};
+  }
+
+  // Retrieve the tile sizes from the last two tiling levels (parallel and
+  // reduction) used for vectorization.
+  SmallVector<int64_t> canonicalVectorShape =
+      mlir::iree_compiler::getTileSizes(rootOp.value(), numTileLevels - 2);
```

Comment:
This is mostly a convention and is not set in stone. Do you have an idea of how to codify this better?

Reply:
Well, this is similar to when we use this information to compute tiles sizes for other purposes. It's a convention but that's what how we have been using it. I guess this will get better once we replace/evolve `lowering_config`?

Reply:
I think the better solution is not to separate the tiling sizes for parallel and reduction. This should happen when we migrate to transform dialect. The lowering config can have a single list for parallel and reduction tiling sizes. When constructing the transform IR, we can simply look into the the matched ops, can create two transform tiling op.